// @wix/cli/build/PreviewCommand-PMRWVEXH.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useCreatePreview
} from "./chunk-FHQ23JQX.js";
import {
  DeploymentStatus
} from "./chunk-6OV37XZP.js";
import "./chunk-TXS6FIYJ.js";
import "./chunk-7LCZ57SO.js";
import "./chunk-2ODMSMLQ.js";
import "./chunk-UBVEF7HP.js";
import "./chunk-AV2QWD2T.js";
import "./chunk-JEYKAALI.js";
import "./chunk-Y7PQOTYY.js";
import "./chunk-MVBGFP6D.js";
import "./chunk-E7EAMFQD.js";
import "./chunk-FUDRK2JS.js";
import "./chunk-OHMQSHHI.js";
import "./chunk-IUJ26DW6.js";
import {
  ErrorViewer
} from "./chunk-KONVQLDI.js";
import "./chunk-WZOSCHKD.js";
import {
  Trans
} from "./chunk-N2KKXUSW.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Badge,
  Box_default,
  Link,
  Spinner,
  useAsync
} from "./chunk-AHOU27PT.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-IWQIW4GL.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-ZUSQSFTR.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-EW4XNYNT.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site/src/components/PreviewCommand/PreviewCommand.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var PreviewCommand = ({ force }) => {
  const createPreview = useCreatePreview();
  const { status } = useAsync(async () => {
    const { previewUrl, deploymentStatus } = await createPreview();
    if (!force && (0, import_variant.isType)(deploymentStatus, DeploymentStatus.Error)) {
      throw deploymentStatus.deploymentError;
    }
    return { previewUrl, deploymentStatus };
  }, []);
  return (0, import_variant.match)(status, {
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, { text: /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "preview_command.loading" }) }),
    Success: ({ result: { previewUrl, deploymentStatus } }) => /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, (0, import_variant.isType)(deploymentStatus, DeploymentStatus.Error) ? /* @__PURE__ */ import_react.default.createElement(Box_default, { marginBottom: 1, flexDirection: "column" }, /* @__PURE__ */ import_react.default.createElement(ErrorViewer, { error: deploymentStatus.deploymentError })) : null, /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react.default.createElement(Badge, { skin: "success" }, /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "general.success" })), " ", /* @__PURE__ */ import_react.default.createElement(
      Trans,
      {
        i18nKey: "preview_command.success",
        components: [/* @__PURE__ */ import_react.default.createElement(Link, { url: previewUrl })]
      }
    )))
  });
};
export {
  PreviewCommand
};
